File: profile-service/src/main/java/project/profileservice/domain/Badge.java
package project.profileservice.domain;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter @Setter
public class Badge {

    @Id 
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "badge_id")
    private Long id;
    private String name;
    private String image_url;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "badge")
    private List<ProfileBadge> profileBadges = new ArrayList<ProfileBadge>();

    
    /**
     * profileBadges 추가 메서드
     */
    public void addProfileBadge(ProfileBadge profileBadge) {
        this.profileBadges.add(profileBadge);
    }
}

File: profile-service/src/main/java/project/profileservice/domain/Badge.kt
package project.profileservice.domain

import lombok.Getter
import project.profileservice.domain.dto.UpdateBadgeDto
import javax.persistence.*

@Entity
@Getter
class Badge
    (
    var name: String,
    var image_url: String,

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) @Column(name = "badge_id")
    var id: Long? = null,
    @OneToMany(cascade = [CascadeType.ALL], mappedBy = "badge")
    val profileBadges: MutableList<ProfileBadge> = mutableListOf(),

    ) {
    fun addProfileBadge(profileBadge: ProfileBadge) {
        this.profileBadges.add(profileBadge)
    }

    fun updateByDto(badgeDto: UpdateBadgeDto) {
        this.name = badgeDto.name
        this.image_url = badgeDto.image_url
    }
}